packages/jsii-pacmak/lib/targets/dotnet/filegenerator.ts

```diff
@@ -0,0 +1,121 @@
+import {CodeMaker} from "codemaker";
+import {Assembly} from "jsii-spec";
+import path = require('path');
+import xmlbuilder = require('xmlbuilder');
+import logging = require('../../logging');
+
+// Represents a dependency in the dependency tree.
+export class DotNetDependency  {
+    public namespace: string;
+    public packageId: string;
+    public fqn: string;
+    public version: string;
+
+    constructor(namespace: string, packageId: string, fqn: string, version: string) {
+        this.namespace = namespace;
+        this.packageId = packageId;
+        this.fqn = fqn;
+        this.version = version;
+    }
+}
+
+// Generates misc files such as the .csproj and the AssemblyInfo.cs file
+// Uses the same instance of CodeMaker as the rest of the code so that the files get created when calling the save() method
+export class FileGenerator {
+
+    private assm: Assembly;
+    private tarballFileName: string;
+    private code: CodeMaker;
+    private assemblyInfoNamespaces: string[] = ["Amazon.JSII.Runtime.Deputy"];
+
+    // We pass in an instance of CodeMaker so that the files get later saved
+    // when calling the save() method on the .NET Generator.
+    constructor(assm: Assembly, tarballFileName: string, code: CodeMaker) {
+        this.assm = assm;
+        this.tarballFileName = tarballFileName;
+        this.code = code;
+    }
+
+    // Generates the .csproj file
+    public generateProjectFile(dependencies: Map<string, DotNetDependency>) {
+        const packageId: string = this.assm.targets!.dotnet!.packageId;
+
+        const projectFilePath: string = path.join(packageId, `${packageId}.csproj`);
+
+        // Construct XML csproj content.
+        // headless removes the <xml?> head node so that the first node is the <Project> node
+        const rootNode = xmlbuilder.create('Project', {encoding: 'UTF-8', headless: true});
+        rootNode.att("Sdk", "Microsoft.NET.Sdk");
+        const propertyGroup = rootNode.ele("PropertyGroup");
+        propertyGroup.ele("TargetFramework", "netstandard2.0");
+        propertyGroup.ele("GeneratePackageOnBuild", "true");
+        propertyGroup.ele("IncludeSymbols", "true");
+        propertyGroup.ele("IncludeSource", "true");
+        propertyGroup.ele("PackageVersion", this.assm.version);
```

Comment:
put this.assm in a local var to avoid repeated lookup. Same with this.assm.targets.dotnet

Reply:
Done, also refactored some other part